### ISSUE TITLE
Cannot fix vehicle engine/motive damage

### ISSUE THREAD

**Issue by SuperStucco** [MegaMek/mekhq#2247]
In 47.14, and for several prior releases, if I try to fix engine/motive damage on a vehicle in the Repair Bay it stays immobile (not just the listed 0 MP - actual motive damage).  Even under the GM tools for editing damage, I can unclick all the checked-off boxes for motive damage, click OK, and when checked again the boxes are still ticked.  Sometimes this goes away if I save and close MekHQ before trying repairs but not always.

Please check the attached campaign file.  There is a single vehicle present, with engine/motive damage.  In the Hangar, right-click and GM Mode => Edit damage...  Uncheck one or more boxes under Motive damage and OK the dialog.  Go to the Repair Bay tab and handle the Repair Motive System task (GM complete if necessary).  Go back to the Hangar and again go through GM Mode => Edit damage...  All motive damage boxes are still checked.  Uncheck all boxes, and OK the dialog.  Once again go through GM Mode => Edit damage...  All motive damage boxes are still checked.

[Vehicle engine fix.cpnx.gz](https://github.com/MegaMek/mekhq/files/5618256/Vehicle.engine.fix.cpnx.gz)

**Comment by sixlettervariables** [comment]
Can you GM Restore the unit?

**Comment by SuperStucco** [comment]
No, that doesn't help - the motive damage check boxes remain checked .  The only thing I can do at the moment is GM acquire a duplicate and move the crew and tech team over.

**Comment by HammerGS** [comment]
I've seen this as well. I normally get with a unit I've salvaged. It's almost like the status doesn't get updated when re-crewed and repaired. 

I did load this campaign and was able to fix everything, and bring crew to full compliment and it stayed at Crippled. Even with an advance on the day.

**Comment by BLR-IIC** [comment]
I just experienced this bug myself yesterday.  I salvaged a destroyed Saladin hover tank and did the GM Restore.  No matter what I did with the GM powers, which also included manually unchecking the motive damage boxes, I couldn't get its MP restored.  Every time I unchecked the motive damage boxes and left the GM damage edit dialog, the boxes would be re-checked when I re-opened it.

**Comment by sixlettervariables** [comment]
There are a few entity update calls MHQ makes when you repair parts, so it seems likely we've missed one for vehicles.

**Comment by sixlettervariables** [comment]
I'm going to keep investigating this on MHQ's side, but I think this is actually going to be a MM bug:
https://github.com/MegaMek/megamek/blob/be733702b9a10dfc716c70c056c896e8fdb17dbc/megamek/src/megamek/client/ui/swing/UnitEditorDialog.java#L694-L708

**Comment by sixlettervariables** [comment]
Talk about a rabbit hole on the MM side. I'm currently having to add `ignoreCrew` to all of the `get*MP(...)` methods to allow `isPermanentlyImmobile` to function properly for tanks without crews.
